components/tooltip/Tooltip.js

```diff
@@ -55,15 +57,15 @@ const tooltipFactory = () => {
         tooltipSize: 'medium',
       };
 
-      constructor() {
+      constructor(props) {
         super(...arguments);
 
         this.state = {
           active: false,
           position: this.props.tooltipPosition,
         };
 
-        this.tooltipRoot = document.createElement('div');
+        this.tooltipRoot = props.documentObject.createElement('div');
```

Comment:
PP: Get rid of `props` as argument and just access it through `this.props` (like is being done when setting `this.state = {}`)